### DISCUSSION TITLE
Stratifiy는 정말 의미가 있는 걸까?

### DISCUSSION THREAD

**Discussion by chanmuzi** [BoocampMonster/korean-relation-extraction#35] [General]
분류 문제에서 각 class의 분포를 유지할 수 있도록 split하는 stratify 방식은 데이터가 불균형할 때 의미 있는 것으로 알려져 있다.
단순히 생각해보더라도 no_relation이 절반 이상을 차지하며, 나머지 29개의 class에 대해 분포가 불균형한 현 대회의 데이터에 대해서는 stratify를 적용하는 것이 타당해 보인다.

1. 그럼에도 불구하고 가장 처음 단순 split을 통해 train, dev set을 구성하여 실험을 진행했을 때 큰 문제를 발견하지 못했다.
2. 오히려 stratify를 적용하여 train, dev set을 구성한 이후 과적합 문제가 발생했다.
3. 또한 특정 class에 대한 예측이 완전 실패하는 것을 확인할 수 있었다.

사실 주어진 데이터를 어떻게 학습시키고 검증해 볼 것인지에 대해서는 대회 초반부, 혹은 진행 과정에서 확인하는 것이 바람직했을 것이다.
하지만 늦었다고 하더라도 '단순 split'과 'stratify split' 방식이 결과를 예측하는 데 있어서 유의미한 차이를 보이는지 확인할 필요가 있다.

**Comment by Ssunbell** [comment]
<img width="685" alt="스크린샷 2022-12-01 오후 7 22 29" src="https://user-images.githubusercontent.com/97590480/205028245-839551ed-de3a-415a-b3c3-73321b66c6bf.png">

1. 데이터 균형을 유지하여 데이터를 학습시키는게 좋다는 것은 누구나 동의할 수 있는 팩트임.
2. 그럼에도 불구하고 성능이 안나온 이유는 위의 그림에서 잘 보여주고 있다고 생각
3. kfold 안에서도 데이터가 밀집된 형태로 K-fold가 진행되므로 다음부터는 `StratifiedShuffleSplit`를 사용해보자~
